sunpycube/cube/cube_utils.py

```diff
@@ -237,13 +236,12 @@ def reduce_dim(cube, axis, keys):
     indices = range(start, stop, step)
     newdata = cube.data.take(indices, axis=axis)
     newwcs = cube.axes_wcs.deepcopy()
-    if keys.step is not None:
-        newwcs.wcs.cdelt[waxis] *= keys.step
-    if keys.start is not None:
-        start = keys.start
-        newwcs.wcs.crpix[waxis] = 0
-        newwcs.wcs.crval[waxis] = (cube.axes_wcs.wcs.crval[waxis] +
-                                   cube.axes_wcs.wcs.cdelt[waxis] * start)
+
+    wcs_slice_data = [slice(start, stop)]
+    for i in range(axis-1, -1, -1):
+        wcs_slice_data.insert(0, slice(0, cube.data.shape[i]))
```

Comment:
If there are 3 axes, and the axis to be sliced is ```1```, doesn't this line result in a ```wcs_slice_data``` of length of 2?  Should ```wcs_slice_data``` not always have the same length as number of axes?